ovn-tester/cms/ovn_kubernetes/__init__.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Canonical
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+
+from ovn_context import Context
+from ovn_workload import WorkerNode, Cluster
+from ovn_utils import DualStackSubnet
+
+ClusterBringupCfg = namedtuple("ClusterBringupCfg", ["n_pods_per_node"])
+
+
+class ovn_kubernetes(object):
```

Comment:
The reason for the all lower case is that the class name will be computed and the class loaded at runtime ref: https://github.com/ovn-org/ovn-heater/pull/173/files#diff-fd90db2a7b9d7962bf6c9f4bc48ab0664be99e8b03d476e049c6513452e96411R175

We can of course teach that code to convert the end user supplied string to PascalCase.

Thanks for the tip on `object`, old habits hard to eradicate ;)

Reply:
> Thanks for the tip on `object`, old habits hard to eradicate ;)

OTOH, the other class definition in the project do appear to inherit from `(object)` so we might want to keep it for consistency.

Reply:
> The reason for the all lower case is that the class name will be computed and the class loaded at runtime ref

I had some thoughts about this too because I'm always wary of dynamic imports. Since there are limited amount of CMS implementation classes, we could create `Dict[str, CMSClass]` mapping between strings from test definitions and implementation classes. I think it'd be bit more deterministic.
There's only one other object in the cms module, `ClusterBringupCfg` named tuple, but that could be moved to the implementation class as well, as a class-level attribute.

Reply:
Maybe I'm misunderstanding this but we dynamically import the test classes, e.g., `cms.ovn_kubernetes.tests.densty_light`.  There can be quite a few different tests per CMS type.  How would a static mapping work for those?

Reply:
> Thanks for the tip on `object`, old habits hard to eradicate ;)

It's actually my fault, I added all those (that shows how often I write python code :D), sorry.  I'll post a cleanup PR soon.

Reply:
yeah, dynamic test loading would probably have to stay as there are many tests. My suggestion was only aimed at mapping between CMS definition in test scenario
```yaml
cms:
  name: openstack
# or
cms:
  name: ovn-kubernetes
````
and their implementation class.
It's not crucial, but I think it'd be bit easier to read and more explicit. Currently proposed `load_cms` function looks like this:
```python
def load_cms(cms_args):
    cms_name = cms_args["name"]
    mod = importlib.import_module(f"cms.{cms_name}")
    class_name = cms_name.replace("-", "_")
    cls = getattr(mod, class_name)
    return cls(cms_args)
  ```
  And my suggestion would look something like this
  ```python
  CMS_MAPPING = {
      "ovn-kubernetes": cms.ovn_kubernetes.OVNKubernetes,
      "openstack": cms.openstack.Openstack,
  }
def load_cms(cms_args):
    cms_name = cms_args["name"]
    cms_class = CMS_MAPPING.get(cms_name)
    if cms_class is None:
        raise RuntimeError()
    return cms_class(cms_args)

Reply:
Sure, sounds OK to me.

Reply:
Ait, let's make it pretty. I'll most likely rely on an attribute in the imported module to tell us the class name to use rather than maintaining a map in the generic code though, but it should serve the same purpose.